### PR TITLE
Fix method body extraction

### PR DESCRIPTION
**Pull request by andyscott**
This is a fix for method body extractions in the compiler. I've also added some new tests.

### CODE REVIEW

File: core/compiler/src/test/scala/com/fortysevendeg/exercises/compiler/MethodBodyReaderSpec.scala [hunk 1]
```diff
@@ -22,34 +17,99 @@ class MethodBodyReaderSpec extends FunSpec with Matchers with MethodBodyReaderSp
...
+       |  */
+       |def addOne(value: Int) = {
+       |    println("this is the first line")
+       |
+       |
+       |  println("this is the last line")
+       |}
+       """.stripMargin
+
+      println("*" * 20)
```
Comment (dialelo) [inline core/compiler/src/test/scala/com/fortysevendeg/exercises/compiler/MethodBodyReaderSpec.scala]: We can get rid of these debugging statements.

**Comment by dialelo** [comment]
Went ahead and deleted some debugging statements, LGTM!

**Comment by dialelo** [comment]
After a conversation with @raulraja we both agreed that we want to keep the newlines in method bodies. They are often useful for separating chunks of code like in the following example:

```scala
import cats.data.Xor

val right: String Xor Int = Xor.right(5)
right.flatMap(x => Xor.right(x + 1)) should be(Xor.right(res0))

val left: String Xor Int = Xor.left("Something went wrong")
left.flatMap(x => Xor.right(x + 1)) should be(Xor.left(res1))
```

**Comment by andyscott** [comment]
My intention was to keep them but I think I wound up removing them. I
needed to handle them differently for the white space counting.

I'll fix it.

**Pull request merged**